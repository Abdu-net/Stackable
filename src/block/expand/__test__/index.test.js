import { edit, save } from '../index'

describe( 'Expand', () => {
	test( 'block edit matches snapshots', () => {
		const wrapper = edit( {
			isSelected: false,
			attributes: {
				text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
				moreText: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
				moreLabel: 'Show more',
				lessLabel: 'Show less',
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )

	test( 'block save matches snapshots', () => {
		const wrapper = save( {
			isSelected: false,
			attributes: {
				text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
				moreText: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
				moreLabel: 'Show more',
				lessLabel: 'Show less',
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )
} )
