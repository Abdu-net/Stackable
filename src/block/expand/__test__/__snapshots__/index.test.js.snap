// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expand block edit matches snapshots 1`] = `
<Fragment>
  <div
    className={undefined}
  >
    <RichText
      className="ugb-expand-less-text"
      multiline="p"
      onChange={[Function]}
      placeholder="Some short text that can be expanded to show more details."
      value="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
    />
    <RichText
      className="ugb-expand-more"
      formattingControls={
        Array [
          "bold",
          "italic",
          "strikethrough",
        ]
      }
      onChange={[Function]}
      placeholder="Show more"
      tagName="a"
      value="Show more"
    />
  </div>
</Fragment>
`;

exports[`Expand block save matches snapshots 1`] = `
<div>
  <div
    className="ugb-expand-less-text"
  />
  <div
    className="ugb-expand-more-text"
    style={
      Object {
        "display": "none",
      }
    }
  />
  <a
    className="ugb-expand-button"
    href="#"
  >
    <Unknown
      className="ugb-expand-more"
      tagName="span"
      value="Show more"
    />
    <Unknown
      className="ugb-expand-less"
      style={
        Object {
          "display": "none",
        }
      }
      tagName="span"
      value="Show less"
    />
  </a>
</div>
`;
